db/db_impl.cc

```diff
@@ -712,6 +712,7 @@ void DBImpl::PersistStats() {
 bool DBImpl::FindStatsByTime(uint64_t start_time, uint64_t end_time,
                              uint64_t* new_time,
                              std::map<std::string, uint64_t>* stats_map) {
+  if (!new_time || !stats_map) return false;
```

Comment:
assert or return false?

Reply:
it's invalid argument, how about assert and return false?

Reply:
Is this a public API? If so, it is better not to abort the process on assertion failure. Instead, I would prefer return some information back to caller.

Reply:
no, this is not a public API

Reply:
Then I would suggest just simply assert new_time and stats_map are not nullptr if we never expect them to be null.